integration/helpers.go

```diff
@@ -184,6 +184,27 @@ func AssertMatchesCommandError(t *testing.T, expected, actual error) {
 	assert.Equal(t, eErr.Code, aErr.Code)
 }
 
+// AssertMatchesWriteErrorCode asserts error code.
+func AssertMatchesWriteErrorCode(t *testing.T, expected, actual error) {
+	t.Helper()
+
+	//nolint: errorlint // using errors.As does not work for mongo.WriteException
+	aErr, ok := actual.(mongo.WriteException)
```

Comment:
Ignoring linter, this didn't work.

```
	var aErr, eErr mongo.WriteException

	if ok := errors.As(actual, aErr); !ok || len(aErr.WriteErrors) != 1 {
		assert.Equal(t, expected, actual)
		return
	}
```

nor this
```
	var aErr, eErr *mongo.WriteException

	if ok := errors.As(actual, &aErr); !ok || len(aErr.WriteErrors) != 1 {
		assert.Equal(t, expected, actual)
		return
	}
```

Reply:
Both examples provide to `errors.As` a value or a pointer to the pointer.
Please consider something like this (it passed test-integration-pg so I assume it's working):
```go
var aErr, eErr mongo.WriteException

if ok := errors.As(actual, &aErr); !ok || len(aErr.WriteErrors) != 1 {
	assert.Equal(t, expected, actual)
	return
}

if ok := errors.As(expected, &eErr); !ok || len(eErr.WriteErrors) != 1 {
	assert.Equal(t, expected, actual)
	return
}

```

Reply:
yea just saw that :see_no_evil: thanks.